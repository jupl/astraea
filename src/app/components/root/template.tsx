import {AppBar, CircularProgress, Toolbar, Typography} from '@material-ui/core'
import React, {ComponentProps, Suspense} from 'react'
import {BrowserRouter as Router, Link, Route, Switch} from 'react-router-dom'
import styled from 'styled-components'
import {GlobalStyle} from './global'

/**
 * Render root component representing the entire application
 * @return Root component
 */
export const AppRoot = () => (
  <Router>
    <>
      <GlobalStyle />
      <AppBar>
        <Toolbar>
          <ToolbarLink to="/">
            <Typography variant="h6">Application</Typography>
          </ToolbarLink>
        </Toolbar>
      </AppBar>
      <Suspense fallback={<Loading />}>
        <Switch>
          <Route path="/color" exact component={ColorPage} />
          <Route path="/" exact component={HomePage} />
          <Route component={NotFoundPage} />
        </Switch>
      </Suspense>
    </>
  </Router>
)

type ProgressProps = ComponentProps<typeof CircularProgress>
const Loading = styled((p: ProgressProps) => <CircularProgress {...p} />)`
  margin: 0 auto;
  align-self: center;
`

const ToolbarLink = styled(Link)`
  text-decoration: none;
`

const HomePage = React.lazy(async() => {
  const module = await import('../home-page')
  return {default: module.AppHomePage}
})

const NotFoundPage = React.lazy(async() => {
  const module = await import('../not-found-page')
  return {default: module.AppNotFoundPage}
})

const ColorPage = React.lazy(async() => {
  const module = await import('../../../color/components/page')
  return {default: module.ColorPage}
})
