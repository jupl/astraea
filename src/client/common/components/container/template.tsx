import React, {ReactNode} from 'react'

interface Props {
  children: ReactNode
}

/**
 * Render container component
 * @return Decorated component
 */
export const Container = ({children}: Props) => <>{children}</>
