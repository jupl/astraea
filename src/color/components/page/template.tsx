import {IconButton} from '@material-ui/core'
import {NavigateBefore, NavigateNext} from '@material-ui/icons'
import * as React from 'react'
import styled from 'styled-components'
import {GlobalStyle} from './global'

// Text component
const Text = styled.span`
  margin-left: 4px;
  margin-right: 4px;
`

// Gradient overlay component
const Gradient = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`

/** Data for color page component */
export interface Props {
  className?: string
  autoColor: boolean
  selectedColor: string
  nextColor(): void
  previousColor(): void
}

/**
 * Render color page component showing current color and buttons to change
 * @param props Component properties
 * @return Color page component
 */
export function ColorPage({
  autoColor,
  nextColor,
  previousColor,
  selectedColor,
}: Props) {
  return (
    <>
      <GlobalStyle color={selectedColor} />
      <Gradient>
        <IconButton color="primary" onClick={previousColor}>
          <NavigateBefore />
        </IconButton>
        <Text>Auto color cycling is {autoColor ? 'on' : 'off'}</Text>
        <IconButton color="primary" onClick={nextColor}>
          <NavigateNext />
        </IconButton>
      </Gradient>
    </>
  )
}
