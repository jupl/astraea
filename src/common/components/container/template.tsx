import React, {ReactNode} from 'react'

/** Component properties */
export interface Props {
  children: ReactNode
}

/**
 * Render container component
 * @return Decorated component
 */
export const Container = ({children}: Props) => <>{children}</>
