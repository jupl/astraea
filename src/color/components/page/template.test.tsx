import {mount, shallow} from 'enzyme'
import * as React from 'react'
import {Button, ColorPage} from './template'

describe('<ColorPage> Template', () => {
  const props = {
    nextColor: jest.fn(),
    previousColor: jest.fn(),
    selectedColor: 'white',
  }

  it('should render as expected', () => {
    expect(shallow(<ColorPage {...props} />)).toMatchSnapshot()
  })

  it('should invoke events as expected', () => {
    const component = mount(<ColorPage {...props} />)
    const buttons = component.find(Button)
    const previousButton = buttons.at(0)
    const nextButton = buttons.at(1)

    expect(props.previousColor).not.toBeCalled()
    expect(props.nextColor).not.toBeCalled()

    previousButton.simulate('click')
    expect(props.previousColor.mock.calls.length).toBe(1)
    expect(props.nextColor).not.toHaveBeenCalled()

    nextButton.simulate('click')
    expect(props.previousColor.mock.calls.length).toBe(1)
    expect(props.nextColor.mock.calls.length).toBe(1)
  })
})
