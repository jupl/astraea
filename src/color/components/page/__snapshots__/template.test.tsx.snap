// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPage> Template should render as expected 1`] = `
<Fragment>
  <GlobalStyleComponent
    color="white"
    suppressMultiMountWarning={false}
  />
  <styled.div>
    <WithStyles(IconButton)
      color="primary"
      onClick={[MockFunction]}
    >
      <pure(NavigateBeforeIcon) />
    </WithStyles(IconButton)>
    <styled.span>
      Auto color cycling is 
      off
    </styled.span>
    <WithStyles(IconButton)
      color="primary"
      onClick={[MockFunction]}
    >
      <pure(NavigateNextIcon) />
    </WithStyles(IconButton)>
  </styled.div>
</Fragment>
`;

exports[`<ColorPage> Template should render as expected 2`] = `
<Fragment>
  <GlobalStyleComponent
    color="white"
    suppressMultiMountWarning={false}
  />
  <styled.div>
    <WithStyles(IconButton)
      color="primary"
      onClick={[MockFunction]}
    >
      <pure(NavigateBeforeIcon) />
    </WithStyles(IconButton)>
    <styled.span>
      Auto color cycling is 
      on
    </styled.span>
    <WithStyles(IconButton)
      color="primary"
      onClick={[MockFunction]}
    >
      <pure(NavigateNextIcon) />
    </WithStyles(IconButton)>
  </styled.div>
</Fragment>
`;
